File: client/client/Services/IBusinessService.cs
using client.DTO.Business;

namespace client.Services;

public interface IBusinessService
{
    public Task<BusinessResponseDto> GetBusinessByIdAsync(string id);
    public Task<BusinessResponseDto> UpdateBusinessProfileAsync(BusinessUpdateRequestDto request);
    
    public Task<List<BusinessResponseDto>> GetBusinessesAsync();

}

File: server/src/main/java/com/example/server/services/IBusinessService.java
package com.example.server.services;

import com.example.server.dto.business.BusinessResponseDto;
import com.example.server.dto.business.BusinessUpdateRequestDto;
import com.example.server.dto.customer.CustomerLocationRequestResponseDto;

import java.io.IOException;
import java.util.List;

public interface IBusinessService {

    BusinessResponseDto getBusinessById(String id);
    BusinessResponseDto updateAndValidateBusinessProfile(
        BusinessUpdateRequestDto updatedProfile) throws IOException;
    List<BusinessResponseDto> getBusinesses();
}
